build/operator/Dockerfile

```diff
@@ -1,4 +1,25 @@
-FROM quay.io/centos/centos:stream9
+ARG BUILD_ARCH=amd64
+FROM --platform=linux/${BUILD_ARCH} quay.io/centos/centos:stream9 AS builder
+
+RUN dnf install -y tar gzip jq && dnf clean all
+RUN ARCH=$(uname -m | sed 's/x86_64/amd64/') && \
+    GO_VERSION=$(curl -L -s "https://go.dev/dl/?mode=json" | jq -r '.[0].version') && \
+    curl -L "https://go.dev/dl/${GO_VERSION}.linux-${ARCH}.tar.gz" -o go.tar.gz && \
+    tar -C /usr/local -xzf go.tar.gz && \
+    rm go.tar.gz
+ENV PATH=$PATH:/usr/local/go/bin
+WORKDIR /go/src/cluster-network-addons-operator
+COPY . .
+
+ARG TARGETOS
+ARG TARGETARCH
+ARG TARGETPLATFORM
```

Comment:
Is this arg used?

Reply:
Yes

Reply:
I may be blind, but where?

Reply:
we use at https://github.com/kubevirt/cluster-network-addons-operator/blob/106c2b661dff49532a0e2b5738f7c12be41a48f2/build/operator/Dockerfile#L18-L19

Reply:
TARGETPLATFORM can be removed.